Formula/gromacs.rb

```diff
@@ -25,7 +25,10 @@ class Gromacs < Formula
   depends_on "gcc" # for OpenMP
   depends_on "openblas"
 
-  fails_with :clang
```

Comment:
I would be happy to test again, compiling with clang. It seems like it would be nice to remove this broad clang-avoidance. But what is the mechanism for choosing compiler toolchain for either `brew install -s gromacs` or for the built-package repositories?

Reply:
*update*:
* I tried `brew install --cc llvm_clang -s gromacs`, but I don't seem to be set up for that.
* I tried `brew install --cc clang -s gromacs` but brew still invoked CMake with `cmake -S . -B build -DGROMACS_CXX_COMPILER=/usr/local/opt/gcc/bin/g++-13 -DGMX_VERSION_STRING_OF_FORK=Homebrew -DGMX_INSTALL_LEGACY_API=ON`